applications/DEMApplication/custom_constitutive/DEM_KDEM_with_damage_CL.cpp

```diff
@@ -0,0 +1,131 @@
+// System includes
+#include <string>
+#include <iostream>
+#include <cmath>
+
+// Project includes
+#include "DEM_KDEM_with_damage_CL.h"
+#include "custom_elements/spheric_continuum_particle.h"
+
+namespace Kratos {
+
+    void DEM_KDEM_with_damage::Initialize(SphericContinuumParticle* element) {}
+
+    DEMContinuumConstitutiveLaw::Pointer DEM_KDEM_with_damage::Clone() const {
+        DEMContinuumConstitutiveLaw::Pointer p_clone(new DEM_KDEM_with_damage(*this));
+        return p_clone;
+    }
+
+    void DEM_KDEM_with_damage::SetConstitutiveLawInProperties(Properties::Pointer pProp, bool verbose) const {
+        KRATOS_INFO("DEM") << "Assigning DEM_KDEM_with_damage to Properties " << pProp->Id() << std::endl;
+        pProp->SetValue(DEM_CONTINUUM_CONSTITUTIVE_LAW_POINTER, this->Clone());
+        this->Check(pProp);
+    }
+
+    void DEM_KDEM_with_damage::CalculateTangentialForces(double OldLocalElasticContactForce[3],
+            double LocalElasticContactForce[3],
+            double LocalElasticExtraContactForce[3],
+            double LocalCoordSystem[3][3],
+            double LocalDeltDisp[3],
+            const double kt_el,
+            const double equiv_shear,
+            double& contact_sigma,
+            double& contact_tau,
+            double indentation,
+            double calculation_area,
+            double& failure_criterion_state,
+            SphericContinuumParticle* element1,
+            SphericContinuumParticle* element2,
+            int i_neighbour_count,
+            bool& sliding,
+            int search_control,
+            DenseVector<int>& search_control_vector,
+            const ProcessInfo& r_process_info) {
+
+        KRATOS_TRY
+
+        const double mTauZero = 0.5 * 1e6 * (GetTauZero(element1) + GetTauZero(element2));
+        const double mInternalFriction = 0.5 * (GetInternalFricc(element1) + GetInternalFricc(element2));
+        const double damage_energy = 0.5 * (element1->GetProperties()[SHEAR_ENERGY_COEF] + element2->GetProperties()[SHEAR_ENERGY_COEF]);
```

Comment:
It would be great if we add a check for this variable. If the variable is not in Properties, at least throw a warning and set a 0.0. That's what we are doing in all constitutive laws lately.

Reply:
Done